pkg/client/request.go

```diff
@@ -478,6 +483,24 @@ func (r *Request) Do() Result {
 			continue
 		}
 
+		if resp.StatusCode == http.StatusServiceUnavailable {
+			if retries == 10 {
+				// Give up after about 10 failed attempts.
+				return Result{respBody, created, err, r.codec}
+			}
+			retries++
+			if waitFor := resp.Header.Get("Retry-After"); waitFor != "" {
```

Comment:
If there's no Retry-After, I think you should give up, because that could indicate a permanent condition. The RFC says:
```
If known, the length of the delay MAY be indicated in a Retry-After header. If no Retry-After is given, the client SHOULD handle the response as it would for a 500 response.
```